### DISCUSSION TITLE
Pulls are being marked as closed instead of merged when pushing and deleting the branch

### DISCUSSION THREAD

**Discussion by skuark** [community/community#70017] [Pull Requests]
### Select Topic Area

Bug

### Body

In my company day-to-day workflow, we usually merge PRs locally, e.g.:

```
$ git checkout main
$ git merge topic-branch
$ git push origin main :topic-branch
```

This way, we push the changes to the main branch, and at the same time we resolve the associated PR and delete the topic branch related to the PR.

Since yesterday, when we run the `git push origin main :topic-branch` command it's happening that the PR is being marked as `Closed` instead of `Merged`, and the PR's related Issues are not being closed automatically neither.

Does anything have changed in GitHub recently to not support this workflow anymore?

**Comment by moenn12** [comment]
@skuark Did you find a fix for this? Having the same issue, Thanks in advance

**Comment by ridvanuyn** [comment]
Hey! We faced same issue and you can reproduce it like that, 

1. Open a pr 
2. Close via time issue
3. Reopen the branch 
4. Merge it 

Then it will shows closed on last state instead of merged
